### PR TITLE
Validating deployment

### PR DESCRIPTION
**Pull request by nikhiljindal**
Adding logic to validate deployments.
Also realized that Deployment.Spec.Replicas does not need to be a pointer in internal API. Changing that to int.

cc @bgrant0607 @ghodss @wojtek-t @lavalamp 

Ref: #1743

### CODE REVIEW

**Comment by wojtek-t** [comment]
LGTM

**Comment by ncdc** [comment]
cc @ironcladlou

File: pkg/expapi/validation/validation.go [hunk 1]
```diff
@@ -53,6 +60,67 @@ func validateHorizontalPodAutoscalerSpec(autoscaler expapi.HorizontalPodAutoscal
 	return allErrs
 }
 
+func ValidatePositiveIntOrPercent(intOrPercent util.IntOrString, fieldName string) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	if intOrPercent.Kind == util.IntstrString {
+		if !util.IsValidPercent(intOrPercent.StrVal) {
```
Comment (ironcladlou) [inline pkg/expapi/validation/validation.go]: Should this be bounded between 0-100 during validation?

**Comment by bprashanth** [comment]
Please re-add lgtm upon addressing dangling comment (and rebasing)

File: pkg/expapi/validation/validation.go (hunk 1 above)
Comment (nikhiljindal) [inline pkg/expapi/validation/validation.go]: I was thinking about that and decided to leave it.
Users specifying a surge of 200% while increasing replicas from 10 to 20 seemed fine to me.
I am validating that its > 0.

But yes for maxUnavailable, we can check that its not more than 100%. Will add that. Thanks!

Comment (nikhiljindal) [inline pkg/expapi/validation/validation.go]: Added that check

**Comment by nikhiljindal** [comment]
Rebased and updated code as per comments.
PTAL

**Comment by ironcladlou** [comment]
LGTM  :+1:

**Comment by nikhiljindal** [comment]
Adding the LGTM tag as per LGTM above.

**Comment by nikhiljindal** [comment]
@k8s-bot ok to test

**Pull request merged**